src/libraries/Native/Unix/configure.cmake

```diff
@@ -30,7 +30,8 @@ endif ()
 # We compile with -Werror, so we need to make sure these code fragments compile without warnings.
 # Older CMake versions (3.8) do not assign the result of their tests, causing unused-value errors
 # which are not distinguished from the test failing. So no error for that one.
-set(CMAKE_REQUIRED_FLAGS "-Werror -Wno-error=unused-value")
+# For clang-5.0 avoid errors like "unused variable 'err' [-Werror,-Wunused-variable]".
+set(CMAKE_REQUIRED_FLAGS "-Werror -Wno-error=unused-value -Wno-error=unused-variable")
```

Comment:
I think that adding the -Wno-error=unused-variable would break the detection for clang older than 5.0 that don't have such a warning . I would prefer to keep it working for that too.

Reply:
Would it be better to disable warnings for the configuration? This is not the first time that the high warning level is causing the configuration to misbehave.

Reply:
I think warnings enabled to catch situations, when expected data types do not match actual ones (for example, `int` is expected, but `long' is actually present).  May be better is to prepare exact list of warnings instead of `-Wall`?

Reply:
We've actually added the -Werror intentionally in the past to fail a detection in case of a "deprecated" warning, see https://github.com/dotnet/corefx/pull/11108.

Reply:
Is the root cause of all this that libraries/native compiles with `-Weverything` ? We have wasted so much time fighting with it.

Reply:
I have tested option `-Wno-error=unused-variable` with **clang-3.9** and **gcc-4.8** -- it works as expected:
```
sysop@jupiter:/tmp$ clang-3.9 --version
clang version 3.9.1-19ubuntu1 (tags/RELEASE_391/rc2)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
sysop@jupiter:/tmp$ clang-3.9 test.cpp -Wall -Werror -Wno-error=unused-value
test.cpp:6:9: error: unused variable 'i' [-Werror,-Wunused-variable]
    int i = rand();
        ^
1 error generated.
sysop@jupiter:/tmp$ clang-3.9 test.cpp -Wall -Werror -Wno-error=unused-value -Wno-unused-variable
sysop@jupiter:/tmp$ 
sysop@jupiter:/tmp$ g++-4.8 test.cpp -Wall -Werror -Wno-error=unused-value
test.cpp: In function ‘int main()’:
test.cpp:6:9: error: unused variable ‘i’ [-Werror=unused-variable]
     int i = rand();
         ^
cc1plus: all warnings being treated as errors
sysop@jupiter:/tmp$ g++-4.8 test.cpp -Wall -Werror -Wno-error=unused-value -Wno-unused-variable
sysop@jupiter:/tmp$ 
sysop@jupiter:/tmp$ cat test.cpp 

#include <stdlib.h>

int main()
{
    int i = rand();
    return 0;
}
```

So, I think inclusion of `-Wno-unused-variable` will not caused new issues and my  commit might be merged.

Reply:
Also instead of `-Wno-unused-variable` might be used option `-Wno-error=unused-variable` -- it works for both, gcc-4.8 and clang-3.9 too:
```
sysop@jupiter:/tmp$ clang-3.9 --version
clang version 3.9.1-19ubuntu1 (tags/RELEASE_391/rc2)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
sysop@jupiter:/tmp$ clang-9 test.cpp -Wall -Werror -Wno-error=unused-value 
test.cpp:6:9: error: unused variable 'i' [-Werror,-Wunused-variable]
    int i = rand();
        ^
1 error generated.
sysop@jupiter:/tmp$ clang-9 test.cpp -Wall -Werror -Wno-error=unused-value -Wno-error=unused-variable
test.cpp:6:9: warning: unused variable 'i' [-Wunused-variable]
    int i = rand();
        ^
1 warning generated.
sysop@jupiter:/tmp$ clang-3.9 test.cpp -Wall -Werror -Wno-error=unused-value 
test.cpp:6:9: error: unused variable 'i' [-Werror,-Wunused-variable]
    int i = rand();
        ^
1 error generated.
sysop@jupiter:/tmp$ clang-3.9 test.cpp -Wall -Werror -Wno-error=unused-value -Wno-error=unused-variable
test.cpp:6:9: warning: unused variable 'i' [-Wunused-variable]
    int i = rand();
        ^
1 warning generated.
sysop@jupiter:/tmp$ gcc-4.8 test.cpp -Wall -Werror -Wno-error=unused-value 
test.cpp: In function ‘int main()’:
test.cpp:6:9: error: unused variable ‘i’ [-Werror=unused-variable]
     int i = rand();
         ^
cc1plus: all warnings being treated as errors
sysop@jupiter:/tmp$ gcc-4.8 test.cpp -Wall -Werror -Wno-error=unused-value -Wno-error=unused-variable
test.cpp: In function ‘int main()’:
test.cpp:6:9: warning: unused variable ‘i’ [-Wunused-variable]
     int i = rand();
         ^
```

Reply:
@jkotas @janvorli   I think my changes might be merged as is (see above).